proto/substrait/algebra.proto

```diff
@@ -178,6 +178,39 @@ message JoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+// A time-series variant of the multi JOIN relational operator left-join-right(s), which joins on an ordered key using inexact matching
+message AsOfJoinRel {
+  RelCommon common = 1;
+  Rel left = 2;
+  repeated Rel right = 3;
+  Expression expression = 4;
```

Comment:
What does this expression mean?

Reply:
That would be the actual join expression. The FieldRef that's still in the proposal right now does not work the way you think it works in Substrait and is just not applicable for a join. See my first few notes on AsOf join in https://github.com/substrait-io/substrait/pull/331#issuecomment-1248529146. I think @rtpsw and me had already aligned on a number of changes to this proposal, but they haven't updated it yet.